### PR TITLE
fix cache mode spine animation playback issue.

### PR DESCRIPTION
**Pull request by Canvasfull**
Re: #

### Changelog

*fix cache mode spine animation playback issue.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/spine/skeleton-cache.ts [hunk 1]
```diff
@@ -69,18 +78,39 @@ export class AnimationCache {
     protected _state: spine.AnimationState = null!;
     protected _skeletonData: spine.SkeletonData = null!;
     protected _skeleton: spine.Skeleton = null!;
-    protected _frames: AnimationFrame[] = [];
+    /**
+     * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
```
Comment (SantyWang) [inline cocos/spine/skeleton-cache.ts]: No need to add `@deprecated`, just mark it `@engineInternal `

File: cocos/spine/skeleton-cache.ts [hunk 2]
```diff
@@ -182,22 +215,168 @@ export class AnimationCache {
...
+            this._enableCacheAttachedInfo = true;
+            this.invalidAllFrame();
+        }
+    }
+
+    // Clear texture quote.
+    public clear () {
+        this._inited = false;
+        for (let i = 0, n = this.frames.length; i < n; i++) {
+            const frame = this.frames[i];
```
Comment (SantyWang) [inline cocos/spine/skeleton-cache.ts]: Nothing changed?

File: cocos/spine/skeleton.ts [hunk 3]
```diff
@@ -270,6 +294,9 @@ export class Skeleton extends UIRenderer {
     constructor () {
         super();
         this._useVertexOpacity = true;
+        this._startEntry = { animation: { name: '' }, trackIndex: 0 } as any;
```
Comment (SantyWang) [inline cocos/spine/skeleton.ts]: Why need a `startEntry`?

File: cocos/spine/skeleton.ts [hunk 4]
```diff
@@ -717,25 +742,39 @@ export class Skeleton extends UIRenderer {
      * @param loop @en Use loop mode or not. @zh 是否使用循环播放模式。
      */
     public setAnimation (trackIndex: number, name: string, loop?: boolean): spine.TrackEntry | null {
+        if (!(typeof name === 'string')) {
+            logID(7509, 'null.');
```
Comment (SantyWang) [inline cocos/spine/skeleton.ts]: Should tell user that it's a invalid input.

File: cocos/spine/skeleton.ts [hunk 5]
```diff
@@ -717,25 +742,39 @@ export class Skeleton extends UIRenderer {
...
-            if (this._cacheMode === AnimationCacheMode.PRIVATE_CACHE) {
-                cache.invalidAnimationFrames();
+            if (cache) {
+                this._animationName = name;
+                this._isAniComplete = false;
+                this._accTime = 0;
+                this._playCount = 0;
+                this._animCache = cache;
+                if (this._socketNodes.size > 0) {
+                    this._animCache.enableCacheAttachedInfo();
```
Comment (SantyWang) [inline cocos/spine/skeleton.ts:774]: What does this function do?

File: cocos/spine/skeleton.ts (hunk 3 above)
Comment (Canvasfull) [reply cocos/spine/skeleton.ts]: This is the simulated start event data, in cache mode.

Comment (SantyWang) [reply cocos/spine/skeleton.ts]: Don't use `any`. You should define an interface.

File: cocos/spine/skeleton-cache.ts (hunk 1 above)
Comment (Canvasfull) [reply cocos/spine/skeleton-cache.ts]: fix and commit.

File: cocos/spine/skeleton.ts (hunk 4 above)
Comment (Canvasfull) [reply cocos/spine/skeleton.ts]: fix and commit.

File: cocos/spine/skeleton-cache.ts (hunk 2 above)
Comment (Canvasfull) [reply cocos/spine/skeleton-cache.ts]: It is invalid code that has been removed and tested.

**Comment by SantyWang** [comment]
@cocos-robot run test cases

**Review by SantyWang — APPROVED** [review]

**Pull request merged**